### ISSUE TITLE
Undelete does nothing when using DELETED_INVISIBLE

### ISSUE THREAD

**Issue by sbussetti** [makinacorpus/django-safedelete#41]
This is actually a pre-existing one I've been running into on Django 1.7/1.8.

When attempting to undelete (save) a deleted model, the `_base_manager` is used by Django to create the "base_qs" to perform the update from:

```python
base_qs = cls._base_manager.using(using)
...
updated = self._do_update(base_qs, using, pk_val, values, update_fields, forced_update)
```

Because the _base_manager is the SafeDeleteManager, only model instances where deleted__isnull=True are considered. This returns and empty set of course because we're attempting to save an instance where deleted is not null.

As a result, `_do_update` returns False, since no records were updated, and django then attempts to insert a new record, which defeats the purpose of undeleting.

So far, the only workaround I've found is to set `_base_manager = models.Manager()`, but this has some other negative side-effects, such as causing reverse managers from instances with ForeignKeys to the SafeDelete model to not filter out deleted models.

**Comment by Gagaro** [comment]
The easiest way would be to add `_do_update` to the SafeDeleteMixin and just call `super` with a different  `base_qs`. I won't have time to test this until tuesday though.

**Comment by Gagaro** [comment]
Also, we should add a test to ensure this is solved on every version.

**Comment by Gagaro** [comment]
I did some tests and the `_base_manage`r is actually the Django one. And these tests should fail if there is a problem. Do you overwrite the `_base_manager` manually on your models?

https://github.com/makinacorpus/django-safedelete/blob/master/safedelete/tests.py#L148-L169

**Comment by sbussetti** [comment]
I'll do a little more research and commit some tests that demonstrate the issue.  I actually *have* to override _base_manager to set it back to models.Manager.

**Comment by rromanchuk** [comment]
Hey guys quick question, documentation says `DELETED_INVISIBLE` is the default, yet i'm still able to retrieve a soft deleted object, am i missing something here? 

```
>>> user = User.users.get(pk=user.guid)
>>> user.deleted
True
```

**Comment by Gagaro** [comment]
Hi @rromanchuk, what version are you using and how is your model configured?

It works fine with the 3.2 and the default `SoftDeleteMixin`:

```
>>> User.objects.create(name='toto')
<User: User object>
>>> User.objects.get()
<User: User object>
>>> User.objects.get().pk
1
>>> a = User.objects.get()
>>> a.delete()
>>> a.deleted
True
>>> User.objects.all()
[]
>>> User.objects.get(pk=1)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/gagaro/.virtualenvs/django1.8-python3.4/lib/python3.4/site-packages/safedelete/managers.py", line 72, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/home/gagaro/.virtualenvs/django1.8-python3.4/lib/python3.4/site-packages/django/db/models/query.py", line 334, in get
    self.model._meta.object_name
models_test.models.DoesNotExist: User matching query does not exist.
>>> User.objects.get(pk=1)
```

**Comment by rromanchuk** [comment]
@Gagaro  it's probably just user error, let me try again with fresh eyes.  Here is what I have though. 

```python
my_mixin = safedelete_mixin_factory(policy=SOFT_DELETE)
class User(my_mixin):
```

with requirements.txt 
`django-safedelete==0.3.2`

**Comment by Gagaro** [comment]
Hi @rromanchuk, do you still have the problem?

**Comment by Gagaro** [comment]
Reopen this issue if you have more info/can reproduce in tests.
